theano/tensor/basic.py

```diff
@@ -4584,9 +4589,34 @@ def c_code(self, node, name, inputs, outputs, sub):
         }
         """ % locals()
 
+def is_flatten(node, outdim=1):
+    return isinstance(node.op, theano.tensor.Reshape) and\
+        node.inputs[1].ndim == outdim
+
 
 def flatten(x, outdim=1):
-    return Flatten(outdim)(x)
+    outdim = int(outdim)
+    # an error is raised if outdim is not positive or
+    # of outdim is greater than one and also greater than x dimensionality.
+    # Any input variable can be flattened to have outdim of 1,
```

Comment:
Great! I'd remove the first two lines, I find them a little cryptic, and I'd add something related after your second sentence, giving roughly
> Any input variable can be flattened to have `outdim` of 1, even if it's a scalar. Otherwise, `outdim` must be positive and smaller than `x.ndim`.